pyincore_incubator/analyses/electricpowernetworkrecovery/electricpowernetworkrecovery.py

```diff
@@ -0,0 +1,793 @@
+
+    def get_spec(self):
+        """Get specifications of the power network analysis.
+
+        Returns:
+            obj: A JSON object of specifications of the power network analysis result.
+
+        """
+        return {
+            "name": "powernetworkanalysis",
+            "description": "Power Network Analysis for damage and recovery",
+            "input_parameters": [
+                {
+                    "id": "result_name",
+                    "required": True,
+                    "description": "Result dataset name",
+                    "type": str,
+                },
+                {
+                    "id": "num_thread",
+                    "required": False,
+                    "description": "Number of threads to use",
+                    "type": int,
+                },
+                {
+                    "id": "num_samples",
+                    "required": False,
+                    "description": "Number of samples to run",
+                    "type": int,
+                },
+                {
+                    "id": "num_crews",
+                    "required": False,
+                    "description": "Number of crews for power pole restoration",
+                    "type": int,
+                },
+                {
+                    "id": "time_step",
+                    "required": False,
+                    "description": "Time step for recovery analysis",
+                    "type": int,
+                },
+                {
+                    "id": "consider_protective_devices",
+                    "required": True,
+                    "description": "Consider protective devices in the analysis",
+                    "type": bool,
+                },
+            ],
+            "input_datasets": [
+                {
+                    "id": "nodes",
+                    "required": True,
+                    "description": "Node dataset",
+                    "type": "incore:epfVer2",
+                },
+                {
+                    "id": "edges",
+                    "required": True,
+                    "description": "Edge dataset",
+                    "type": "incore:epfVer2",
+                },
+                {
+                    "id": "buildings",
+                    "required": True,
+                    "description": "Building dataset",
+                    "type": "ergo:buildingInventoryVer7",
+                },
+                {
+                    "id": "pole_damage",
+                    "required": True,
+                    "description": "Pole damage samples dataset",
+                    "type": "incore:sampleFailureState",
+                },
+            ],
+            "output_datasets": [
+                {
+                    "id": "result",
+                    "parent_type": "buildings",
+                    "description": "GeoDataFrame of power back time results",
+                    "type": "ergo:buildingPowerRecoveryVer1",
```

Comment:
Since this is a new type coming from the IN-CORE project, can you call this incore:buildingPowerRecovery or incore:buildingPowerRecoveryVer1?